### PR TITLE
DAOS-12723 test: fix ExecutableCommand.with_exports for sudo

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: pr,vm container_acl,vm DfuseMUPerms
Test-tag: dfuse,pr DfuseMUPerms
Skip-unit-tests: true
Skip-fault-injection-test: true

Instead of:
`export foo=bar; sudo <cmd>`
The correct way is:
`sudo foo=bar <cmd>`

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by daltonbohning** [comment]
Feedback from Phil:
```
str(self)       echo hello
    with_bind       taskset -c 1 echo hello
    with_sudo       sudo taskset -c 1 echo hello
    with_exports    sudo foo=bar taskset -c 1 echo hello
```

**Comment by daltonbohning** [comment]
To prove the exports with sudo work, consider this weekly run that uses
```
export D_LOG_FILE=/var/tmp/daos_testing/dfuse_daos.log;export D_LOG_MASK=INFO;export COVFILE=/tmp/test.cov; sudo -n runuser -u daos_test_user_x1 -- /usr/bin/dfuse /tmp/daos_dfuse_test_dfuse_mu_perms TestLabel_1 F220C479-4AAA-42E8-93B5-6E1350AB7808 --disable-caching --multi-user
```
And does not have a dfuse_daos.log:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/master/8171/artifact/Functional%20on%20EL%208/dfuse/mu_perms.py/daos_logs.wolf-232vm3/

But with this PR, which uses:
```
sudo D_LOG_FILE=/var/tmp/daos_testing/dfuse_daos.log D_LOG_MASK=INFO COVFILE=/tmp/test.cov -n runuser -u daos_test_user_x1 -- /usr/bin/dfuse /tmp/daos_dfuse_test_dfuse_mu_perms TestLabel_1 F70881E5-AE4F-49EA-BC13-3CCB380966C7 --disable-caching --multi-user
```
There is now a dfuse_daos.log:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-11570/5/artifact/Functional%20on%20EL%208/dfuse/mu_perms.py/daos_logs.wolf-105vm3/

File: src/tests/ftest/util/run_utils.py [hunk 1]
```diff
@@ -162,23 +163,6 @@ def log_output(self, log):
                     log.debug("    %s", line)
 
 
-def get_switch_user(user="root"):
```
Comment (daltonbohning) [inline src/tests/ftest/util/run_utils.py:165]: I removed this, as it doesn't make sense in the context of environments

**Comment by daltonbohning** [comment]
Existing failures in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-11570/9/tests
- FTEST_daos_test.DAOS_Rebuild_EC
  - https://daosio.atlassian.net/browse/DAOS-12709
- soak/smoke.py
  - similar to https://daosio.atlassian.net/browse/DAOS-12706

**Review by shimizukko — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]
I assume all related tests are passed.

**Review by jamesanunez — APPROVED** [review]

**Pull request merged**